metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/CacheConfig.java

```diff
@@ -0,0 +1,74 @@
+package com.linkedin.gms.factory.common;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.EvictionConfig;
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MaxSizePolicy;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spring.cache.HazelcastCacheManager;
+import java.util.concurrent.TimeUnit;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.caffeine.CaffeineCacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+
+@Configuration
+public class CacheConfig {
+
+  @Value("${CACHE_TTL_SECONDS:600}")
+  private int cacheTtlSeconds;
+
+  @Value("${CACHE_MAX_SIZE:10000}")
+  private int cacheMaxSize;
+
+  @Value("${searchService.cache.hazelcast.serviceName:hazelcast-service}")
+  private String hazelcastServiceName;
```

Comment:
why do we have hazelcast in the name of variables here? 
isn't the idea that we can swap in differnt impls for the caffeine cache? 
does this seem overly-specific?

Reply:
i see - so we are only accounting for 2 cases: caffeine and hazel. i think i get it

Reply:
Spring Cache Manager is intended for very very simple use cases when switching out cache implementations. Unfortunately our use cases are complex enough where it requires using the provider level interfaces so we would need to reimplement any other implementations we support. This is unchanged from how it previously was, but just exposes the implementation name in the config. It's possible that other providers would also require a headless K8s service for their distributed cache implementation, but unlikely (if we even do implement another underlying supported cache) so I think it's okay to be specific here.